config/electrum.go

```diff
@@ -63,11 +70,20 @@ func (c *Config) resolveElectrum() error {
 
 	// #nosec G404 (insecure random number source (rand))
 	// Picking up an Electrum server does not require secure randomness.
-	selectedURL := urls[rand.Intn(len(urls))]
+	selectedURL := urls[rng.Intn(len(urls))]
 
 	// Set only the URL in the original config. Other fields may be already set,
 	// and we don't want to override them.
 	c.Bitcoin.Electrum.URL = selectedURL
 
+	// Blockstream's servers timeout session after 60 seconds of inactivity which
+	// is much shorter than expected 600 seconds. To workaround connection drops
+	// and logs pollution with warning we reduce the KeepAliveInterval for the
+	// Blockstream's servers to less than 60 seconds.
+	if c.Bitcoin.Electrum.KeepAliveInterval == 0 &&
+		strings.Contains(selectedURL, "electrum.blockstream.info") {
+		c.Bitcoin.Electrum.KeepAliveInterval = blockstreamKeepAliveInterval
+	}
```

Comment:
This is good for now but ultimately, I'd rather replace `config/_electrum_urls/mainnet` file with `config/electrum/mainnet` directory and separate config files for each electrum server.